jwst/wfss_contam/tests/test_observations.py

```diff
@@ -0,0 +1,112 @@
+import pytest
+import numpy as np
+import asdf
+import os
+
+from astropy.convolution import convolve
+from photutils.segmentation import make_2dgaussian_kernel
+from astropy.stats import sigma_clipped_stats
+from photutils.datasets import make_100gaussians_image
+from photutils.segmentation import SourceFinder
+
+from jwst.wfss_contam.observations import background_subtract
+from jwst.wfss_contam.disperse import dispersed_pixel
+from jwst.wfss_contam.tests import data
+from jwst.datamodels import SegmentationMapModel, ImageModel
+
+data_path = os.path.split(os.path.abspath(data.__file__))[0]
+DIR_IMAGE = "direct_image.fits"
+
+
+@pytest.fixture(scope='module')
+def direct_image():
+    data = make_100gaussians_image()
+    kernel = make_2dgaussian_kernel(3, size=5)
+    data = convolve(data, kernel)
+    return data
+
+
+@pytest.fixture(scope='module')
+def direct_image_with_gradient(tmp_cwd_module, direct_image):
+    ny, nx = direct_image.shape
+    y, x = np.mgrid[:ny, :nx]
+    gradient = x * y / 5000.0
+    data = direct_image + gradient
+
+    # obs expects input list of direct image filenames
+    model = ImageModel(data=data)
+    model.save(DIR_IMAGE)
```

Comment:
Thanks for creating the image on the fly, because we try to avoid adding data files to the repo.

Reply:
yeah I figured that's what you would want